samcli/commands/_utils/custom_options/hook_package_id_option.py

```diff
@@ -74,15 +74,26 @@ def handle_parse_result(self, ctx, opts, args):
                 LOG.debug("beta-feature flag is disabled and prepare hook is not run")
                 return super().handle_parse_result(ctx, opts, args)
 
+            iac_project_path = os.getcwd()
+            output_dir_path = os.path.join(iac_project_path, ".aws-sam-iacs", "iacs_metadata")
+
+            # check if user provided --skip-prepare-iac
+            skip_prepare_hook = opts.get("skip_prepare_iac")
+            metadata_file_path = os.path.join(output_dir_path, "template.json")
```

Comment:
question: Is it safe to assume that this template file will always be generated in this location (`.aws-sam-iacs/iacs_metadata`?

Reply:
This will work for terraform case which is OK, but I think we should push it to the Prepare Hook, and the hook itself should do this check. The idea later we may have multiple templates to be generated by the hook (like CDK for example) ... also, the template file name can be different based on how the hook will name it (also CDK)

Reply:
Sounds good, in that case I'll move this check into the prepare hook under the terraform folder.